pep_sphinx_extensions/pep_processor/transforms/pep_headers.py

```diff
@@ -68,19 +68,20 @@ def apply(self) -> None:
                 raise PEPParsingError(msg)
 
             para = body[0]
-            if name in {"author", "bdfl-delegate", "pep-delegate", "discussions-to", "sponsor"}:
+            if name in {"author", "bdfl-delegate", "pep-delegate", "sponsor"}:
                 # mask emails
                 for node in para:
                     if not isinstance(node, nodes.reference):
                         continue
-                    if name == "discussions-to":
-                        if node["refuri"].startswith("http"):
-                            node[0] = _list_name_from_thread(node)
-                        else:
-                            node[0] = _mask_email(node)
-                            node["refuri"] += f"?subject=PEP%20{pep_num}"
-                    else:
-                        node.replace_self(_mask_email(node))
+                    node.replace_self(_mask_email(node))
+            elif name in {"discussions-to", "resolution"}:
+                # only handle threads, email addresses in Discussions-To aren't
+                # masked.
+                for node in para:
+                    if not isinstance(node, nodes.reference):
+                        continue
+                    if node["refuri"].startswith("https://mail.python.org"):
```

Comment:
I'm not sure I follow, it seems like you removed any `https://mail.python.org` link in existing PEPs.

Reply:
I'm not sure I follow what you're not following, sorry. I don't see anywhere the @AA-Turner removed existing thread links, and this is to process those that have been added to this field now and (with the new PEP 1 guidance) in the future, to display the mailing list name and an inline link instead of the very long, unsightly and hard to parse bare link.

Reply:
Perhaps I don't fully understand what this code is doing, but here's my thinking:
- This if statement covers the `Discussions-To` header, and only does something if the text(?) in that header starts with `https://mail.python.org`.
- Everything we're discussing below happens in the `_pretty_thread` function, which is only called here.
- But none of the existing PEPs actually have a header that starts with `https://mail.python.org`, so this doesn't affect any of these links.

Reply:
Oh, that last part isn't true:
```
pep-0613.rst:Discussions-To: https://mail.python.org/archives/list/typing-sig@python.org/thread/MWRJOBEEEMFVXE7CAKO7B4P46IPM4AN3/
```
I'm not clear though on why the Discussions-To headers pointing to lists were also modified in this PR.